cnn_framework/data.py

```diff
@@ -385,19 +385,20 @@ def prepare_full_train_data_from_file(datafile,
 def split_test_from_train_and_val(X, y, extra_data=None, shuffle_seed=None, testing_ratio=0.1):
 
     # Feed shuffle seed
+    rng = np.random.RandomState()
     if shuffle_seed is not None:
-        np.random.seed(shuffle_seed)
+        rng.seed(shuffle_seed)
 
     # Get random number generator state so that we can shuffle multiple arrays
-    rng_state = np.random.get_state()
+    rng_state = rng.get_state()
 
     # Shuffle data in place
-    np.random.shuffle(X)
-    np.random.set_state(rng_state)
-    np.random.shuffle(y)
+    rng.shuffle(X)
```

Comment:
What is the purpose of the changes in this commit? As far as I can tell, the previous code has the same effect.

Reply:
I don't think setting random seed of numpy often is a good idea (it affects globally).  A RandomState is just a local random number generator, which is more suitable here in my opinion.

Reply:
Oh, I see, that makes sense. What do you think about removing the explicit seed assignments? For example, in the calls to `split_test_from_train_and_val` and `prepare_folded_data` in many of the functions in `data.py` and some of the calls in `predictor.py`. Is it desirable to get the same data split each time?